src/utils/rdf.js

```diff
@@ -0,0 +1,83 @@
+import { Parser } from 'n3';
+import createClient from 'ipfs-http-client';
+import { fromJsonldjsNode } from './claimgraph';
+
+const ipfsDefaultConfig = 'http://localhost:5001';
+
+let ipfsClient;
+
+/**
+ * Writes a document to IPFS and returns its CID
+ * @param {string} document - RDF document to store on IPFS
+ * @param {object} connectionConfig - IPFS HTTP Client connection options
+ * @returns {Promise<any>}
+ */
+export async function writeToIPFS(document, connectionConfig = ipfsDefaultConfig) {
+  if (!ipfsClient) {
+    ipfsClient = createClient(connectionConfig);
+  }
+  const { cid } = await ipfsClient.add(document);
+  return cid;
+}
+
+/**
+ * Dereferences a CID from IPFS into a string, expects a running node to be connected to
+ * @param {string} cid - IPFS document cid hash
+ * @param {object} connectionConfig - IPFS HTTP Client connection options
+ * @param {object} options - IPFS HTTP Client options passed to cat
+ * @returns {Promise<any>}
+ */
+export async function dereferenceFromIPFS(cid, connectionConfig = ipfsDefaultConfig, options = {}) {
+  if (!ipfsClient) {
+    ipfsClient = createClient(connectionConfig);
+  }
+
+  const result = [];
+  const document = await ipfsClient.cat(cid, options);
+  for await (const entry of document) {
+    result.push(entry.toString());
+  }
+
+  return result.join('\n');
+}
+
+/**
+ * Parses an RDF document string and formats it according to rify requirements
+ * @param {string} document - RDF document in turtle or other form
+ * @param {object} parserOptions - N3 Parser configuration object
+ * @returns {*}
+ */
+export function parseRDFDocument(document, parserOptions = {}) {
```

Comment:
When crawling we should be strict about the type of document we accept. I think we need a function that will accept only turtle.

My primary concern is that the meaning of a document may change depending on the format a parser assumes. I don't want it to be possible for attestations have multiple interpretations.

Is there are reason this should accept multiple formats?

Reply:
Passing `{ format: "text/turtle" }` to the parser constructor will force turtle.

Reply:
That's what parseroptions are for when you call this method, this method shouldn't force turtle as it may be used not by crawler since we're offering it in the SDK

Reply:
I see. As long as the crawler has the ability to force triples it should be safe.